lib/install/validate/storage.go

```diff
@@ -58,18 +62,56 @@ func (v *Validator) storage(ctx context.Context, input *data.Data, conf *config.
 	}
 
 	// TODO: add volume locations
-	for label, volDSpath := range input.VolumeLocations {
-		dsURL, _, err := v.DatastoreHelper(ctx, volDSpath, label, "--volume-store")
-		v.NoteIssue(err)
+	for label, volPath := range input.VolumeLocations {
+		nfsURL, nfsErr := NFSHelper(volPath)
+		if nfsURL != nil && nfsErr == nil {
+			conf.VolumeLocations[label] = nfsURL
+			continue
+		}
+
+		dsURL, _, dsErr := v.DatastoreHelper(ctx, volPath, label, "--volume-store")
+
+		// if both validations return error then the input cannot be used as a volumestore.
+		if nfsErr != nil && dsErr != nil {
+			v.NoteIssue(nfsErr)
+			v.NoteIssue(dsErr)
+		}
+
 		if dsURL != nil {
 			conf.VolumeLocations[label] = dsURL
 		}
 	}
 }
 
+// NFSHelper is used to construct a potential nfs target. If a valid NFS target cannot be contsructed a nil pointer will be returned.
+func NFSHelper(rawPath string) (*url.URL, error) {
+	// NOTE: issues from this function will need to be noted at the end of the datastore validation attempt. since we have multiple acceptable input formats now.
+	// NOTE: USER is checked at portlayer configuration time, since we cannot validate whether one is needed or not during vic-machine create time.
+
+	u, err := url.Parse(rawPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// XXX: the below cases indicate a successful url parsing path, and should be accounted for in the error output from the validator
+	// at a minimum we expect a proper scheme, a host, and a path. without these a nfs store is not identifiable and able to be targeted
+	if u.Scheme != nfsScheme {
+		return u, fmt.Errorf("volume store target (%s) needs to have scheme 'nfs' if it is intended as an nfs volume-store. format: <nfs://<user>:<password>@<host>/<share point path>:label", rawPath)
```

Comment:
@hickeng and I were talking about this and we have an altenative.

First, `NFS` doesn't have passwords.  The user here is the UID and the password was intended to be the GID ('uid:gid`, ala the args passed to `chown`).  This UID/GID mapping is used to authenticate with the server.  The expectation is the local system has authenticated this user already and passing the uid/gid for him/her is all that the rpc server needs to allow the target to call the mount program.  Normally this is `0:0` when mounting from a host to an nfs server.  We're allowing you (the user) to change this to conform to whatever your environment expects.  We then use this uid/gid on the VCH only (not the container) to create files as this uid/gid (see anonuid in `man 5 nfs`).

So the alternative we have is allowing the user to use `nfs://<host>/path/to/target,comma,separated,mount,options,like,proto=tcp,or,vers=3`.  We can then scan for the `uid=` and `gid=` fields and use them instead of abusing the `User()` and `Password()` sections of the URL struct.